apis/management.cattle.io/v3/pipeline_types.go

```diff
@@ -181,11 +182,12 @@ type PipelineExecutionSpec struct {
 type PipelineExecutionStatus struct {
 	Conditions []PipelineCondition `json:"conditions,omitempty"`
 
-	Commit         string        `json:"commit,omitempty"`
-	ExecutionState string        `json:"executionState,omitempty"`
-	Started        string        `json:"started,omitempty"`
-	Ended          string        `json:"ended,omitempty"`
-	Stages         []StageStatus `json:"stages,omitempty"`
+	Commit          string            `json:"commit,omitempty"`
+	ExecutionState  string            `json:"executionState,omitempty"`
+	Started         string            `json:"started,omitempty"`
+	Ended           string            `json:"ended,omitempty"`
+	Stages          []StageStatus     `json:"stages,omitempty"`
+	OptionalEnvVars map[string]string `json:"optionalEnvVars,omitempty"`
```

Comment:
having *optional* in the name doesn't sound quite right. Lets consider calling it `envVars` or something like that